Engine/Engine/Core/Mesh.cpp

```diff
@@ -17,62 +17,12 @@ Mesh::Mesh()
  * \param (glm::vec3) position - The position of the mesh
  */
 Mesh::Mesh(boost::shared_ptr<MeshData> meshData, boost::shared_ptr<ShaderProgram> sp, boost::shared_ptr<Texture> t, glm::vec3 position) :
-	subMeshData(boost::shared_ptr<MeshData>(meshData)),
```

Comment:
Even if it's going to happen later, or just getting a default value like `nullptr`, it's a good idea to make sure everything at least gets instantiated when using initializer lists like this. Explicit initialization is a good protective measure against a slew of obscure bugs that can crop up down the road as things change.